docs/src/main/sphinx/connector/kafka.rst

```diff
@@ -61,16 +67,17 @@ Configuration properties
 
 The following configuration properties are available:
 
-========================================================== ==============================================================================
+========================================================== ======================================================================================================
 Property Name                                              Description
-========================================================== ==============================================================================
+========================================================== ======================================================================================================
 ``kafka.default-schema``                                   Default schema name for tables
 ``kafka.nodes``                                            List of nodes in the Kafka cluster
 ``kafka.buffer-size``                                      Kafka read buffer size
 ``kafka.hide-internal-columns``                            Controls whether internal columns are part of the table schema or not
 ``kafka.messages-per-split``                               Number of messages that are processed by each Trino split, defaults to 100000
 ``kafka.timestamp-upper-bound-force-push-down-enabled``    Controls if upper bound timestamp push down is enabled for topics using ``CreateTime`` mode
-``kafka.security-protocol``                                Security protocol for connection to Kafka cluster, defaults to ``PLAINTEXT``
+``kafka.security-protocol``                                An optional security protocol for connection to Kafka cluster.
+                                                           If omitted, ``kafka.config.resources`` will be loaded
```

Comment:
`kafka.config.resources` should be loaded always. Explicit configuration from connector properties file should override it. So `kafka.security-protocol` is not more optional, than it used to be.

Reply:
Currently, the logic is following your statement https://github.com/trinodb/trino/pull/8743#discussion_r773780316 

> ...  it should not be overwritten by kafka.config.resource. In case it is not defined, then one from kafka.config.resource should be used.

In other words, when `kafka.security-protocol` is undefined, then the one from `kafka.config.resource` will be used. If both `kafka.security-protocol` and `security.protocol`in `kafka.config.resource` are defined, `kafka.security-protocol` has the priority.

Reply:
I have removed the optional statement.